terraform/graph_config_node_resource.go

```diff
@@ -174,7 +176,7 @@ func (n *GraphNodeConfigResource) DynamicExpand(ctx EvalContext) (*Graph, error)
 // GraphNodeAddressable impl.
 func (n *GraphNodeConfigResource) ResourceAddress() *ResourceAddress {
 	return &ResourceAddress{
-		// Indicates no specific index; will match on other three fields
+		Path:         n.Path[1:],
```

Comment:
Decided that it's the Addressable Node's responsibility to strip off "root" as a prefix to their module path, rather than the Resource Address's responsibility to _prepend_ `"root"` to the module path of every parsed address.